src/twisted/web/_flatten.py

```diff
@@ -12,29 +12,72 @@
 
 from sys import exc_info
 from types import GeneratorType
+from typing import (
+    Any,
+    Callable,
+    Coroutine,
+    Generator,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    TypeVar,
+    Union,
+    cast,
+)
 from traceback import extract_tb
 from inspect import iscoroutine
 
 from twisted.python.compat import nativeString
+
 from twisted.internet.defer import Deferred, ensureDeferred
+from twisted.python.failure import Failure
 from twisted.web._stan import Tag, slot, voidElements, Comment, CDATA, CharRef
 from twisted.web.error import UnfilledSlot, UnsupportedType, FlattenerError
-from twisted.web.iweb import IRenderable
+from twisted.web.iweb import IRenderable, IRequest
+
+
+T = TypeVar("T")
 
+FlattenableRec = Any
+"""
+For documentation purposes, read C{FlattenableRec} as L{Flattenable}.
+However, since mypy doesn't support recursive type definitions (yet?),
+we'll put Any in the actual definition.
+"""
```

Comment:
Yeah I think JSON has this problem also.

Reply:
What's `Rec` stand for?

Reply:
Presumably `Recursive` ?

Reply:
> Yeah I think JSON has this problem also.

Every tree structure has this problem.

> What's `Rec` stand for?

> Presumably `Recursive` ?

Yes, recursive. I'll write it out in full to make it clearer.

Reply:
Cool.  I'm glad that you get some non-nested type checking here as a baseline.